sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql

```diff
@@ -497,12 +507,30 @@ SELECT
     `end` => DATETIME(end_date, plan_interval_timezone),
     grace_period => billing_grace_period,
     inclusive => FALSE
-  ) AS months_retained,
+  ) AS plan_months_retained,
+  mozfun.norm.diff_months(
+    start => DATETIME(
+      COALESCE(original_subscription_start_date, subscription_start_date),
+      plan_interval_timezone
+    ),
+    `end` => DATETIME(end_date, plan_interval_timezone),
+    grace_period => billing_grace_period,
+    inclusive => FALSE
+  ) AS subscription_months_retained,
   mozfun.norm.diff_months(
     start => DATETIME(subscription_start_date, plan_interval_timezone),
     `end` => DATETIME(TIMESTAMP(CURRENT_DATE), plan_interval_timezone),
     grace_period => billing_grace_period,
     inclusive => FALSE
+  ) AS current_months_since_plan_start,
+  mozfun.norm.diff_months(
+    start => DATETIME(
+      COALESCE(original_subscription_start_date, subscription_start_date),
+      plan_interval_timezone
+    ),
+    `end` => DATETIME(TIMESTAMP(CURRENT_DATE), plan_interval_timezone),
+    grace_period => billing_grace_period,
+    inclusive => FALSE
   ) AS current_months_since_subscription_start,
```

Comment:
@sean-rose  The name convention is really a good point. 
I would prefer option 1 as I think it would be important to have `original_subscription_months_retained` and  `current_months_since_original_subscription_start` for the retention/churn metrics in total and  the other two for the retention/churn metrics by plan. i.e. the upgrades should not affect the total churn/retention but the number of each plan group changes like X subscriptions moved from plan A to plan B meaning X churns in plan A due to the upgrade.